### ISSUE TITLE
the problem caused by padding-top of headline

### ISSUE THREAD

**Issue by tirear** [thaider/Tweeki#140]
Original headline (select all): The problem is that user can never click any links in that blue area. Because the mouse is detected as hovering on headline.
![trouble](https://user-images.githubusercontent.com/16443105/54733559-3ae95c00-4bd5-11e9-939b-0fcec5c47a34.png)

==============================This is dividing line ==============================

Modified headline (select all)
![tmpSolution](https://user-images.githubusercontent.com/16443105/54733565-3cb31f80-4bd5-11e9-9f04-e81d0c7319cd.png)

by changing the following lines in file Tweeki/components/tweeki-base.less
From:
.with-navbar.with-navbar-fixed .mw-headline {
        padding-top: 70px;
}

To:
/\*.with-navbar.with-navbar-fixed .mw-headline {
        padding-top: 70px;
}\*/
.with-navbar.with-navbar-fixed {
  padding-top: 70px;
}
.mw-headline {
  padding-top: 0px;
}

or can be replaced by "margin-top"

**Comment by thaider** [comment]
tl;dr: should be fixed in the latest code…

Hi. Thanks for the report and proposed solution. It's a complex issue. The reason for the padding to be introduced in the first place is that without it headlines are hidden under a fixed navbar when following a link to a section:

compare this result after following the link to ```#PageLayout```

![withpadding](https://user-images.githubusercontent.com/3634883/57914831-c0b11d00-788f-11e9-87f7-112be3da5b52.png)

…with this when I implement your proposed changes

![witouthpadding](https://user-images.githubusercontent.com/3634883/57914896-e63e2680-788f-11e9-90f5-97531875a7a5.png)

Of course adding the padding is not a perfect solution as it makes the content above unaccessible. That's the reason why there was a javascript function that changed the structure of the code from 

```<span id="ID" class="mw-headline">Text</span>``` 

to

```<span id="ID" class="mw-headline"></span>Text``` 

This worked fine as long as there was no problem with javascript. One result of a javascript error is the non-execution of all later code. There is an issue with Bootstrap's scrollspy feature that expects ids to be valid jQuery selectors. That often is not the case in MediaWiki because headlines (and their ids) can have special chars and punctuation in it. I forgot to apply the necessary patch after updating Bootstrap to v3.4. 

**It should be working again with the latest code. Could you please confirm this?** 

Additionally I moved the manipulation of the headlines from javascript to server-side code that creates the following structure for headlines and is hopefully more reliable:

```<span id="ID" class="mw-headline"></span><span class="mw-headline-content">Text</span>```

**Comment by tirear** [comment]
The effect has been confirmed. Thanks for your explanation and solution. It's great!

**Comment by bimac** [comment]
Is it possible that the same issue persists on _Special:SpecialPages_? I can't click the links directly above headlines.

**Comment by thaider** [comment]
Thanks for the comment, @bimac . You're absolutely right. It was disabled on special pages because it was causing other problems. In the meanwhile there's a new css property, ```scroll-padding-top``` that can be used without having to rely on ugly hacks. It's fixed in the latest version.

**Comment by bimac** [comment]
Thank you for the update, @thaider - it works fine now!
